src/components/FiltersDrawer/Filter/index.tsx

```diff
@@ -0,0 +1,191 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import Button, { Variant } from "@leafygreen-ui/button";
+import IconButton from "@leafygreen-ui/icon-button";
+import { palette } from "@leafygreen-ui/palette";
+import {
+  SegmentedControlOption as Option,
+  SegmentedControl,
+} from "@leafygreen-ui/segmented-control";
+import TextInput from "@leafygreen-ui/text-input";
+import { Body } from "@leafygreen-ui/typography";
+import Icon from "components/Icon";
+import { size } from "constants/tokens";
+
+const { gray } = palette;
+
+enum CaseSensitivity {
+  Sensitive = "sensitive",
+  Insensitive = "insensitive",
+}
+
+enum MatchType {
+  Exact = "exact",
+  Inverse = "inverse",
+}
+
+interface FilterProps {
+  ["data-cy"]?: string;
+  filterName: string;
+  deleteFilter: (filter: string) => void;
+  editFilter: (oldFilter: string, newFilter: string) => void;
+}
+
+const Filter: React.FC<FilterProps> = ({
+  "data-cy": dataCy,
+  filterName,
+  deleteFilter,
+  editFilter,
+}) => {
+  const [newFilter, setNewFilter] = useState("");
+  const [isEditing, setIsEditing] = useState(false);
+  const [isVisible, setIsVisible] = useState(true);
+  const [caseSensitivity, setCaseSensitivity] = useState<string>(
+    CaseSensitivity.Insensitive
+  );
+  const [matchType, setMatchType] = useState<string>(MatchType.Exact);
+
+  const resetEditState = () => {
+    setIsEditing(false);
+    setNewFilter("");
+  };
+
+  return (
+    <FilterRow data-cy={dataCy}>
+      <FilterLabel>
+        <Badge>FILTER</Badge>
+        <FilterIcons>
+          <IconButton
+            aria-label="Edit filter button"
+            onClick={() => setIsEditing(true)}
+          >
+            <Icon fill={gray.base} glyph="Edit" />
+          </IconButton>
+          <IconButton
+            aria-label="Visibility filter button"
+            onClick={() => setIsVisible(!isVisible)}
+          >
+            <Icon
+              fill={gray.base}
+              glyph={isVisible ? "Visibility" : "ClosedEye"}
+            />
+          </IconButton>
+          <IconButton
+            aria-label="Delete filter button"
+            onClick={() => deleteFilter(filterName)}
+          >
+            <Icon fill={gray.base} glyph="X" />
+          </IconButton>
+        </FilterIcons>
+      </FilterLabel>
+
+      {isEditing ? (
+        <>
+          <StyledTextInput
+            aria-label="Edit Filter Name"
+            data-cy="edit-filter-name"
+            onChange={(e) => setNewFilter(e.target.value)}
+            placeholder="New filter definition"
+            sizeVariant="small"
+            spellCheck={false}
+            type="search"
+            value={newFilter}
```

Comment:
Just flagging based on design review today we should show the existing filter value here for users to edit!